ui/index.d.ts

```diff
@@ -666,18 +685,18 @@ export interface IWizardOptions<T> {
 /**
  * A wizard that links several user input steps together
  */
-export declare class AzureWizard<T extends {}> {
+export declare class AzureWizard<T extends IActionContext> {
     /**
      * @param wizardContext  A context object that should be used to pass information between steps
      * @param options Options describing this wizard
      */
     public constructor(wizardContext: T, options: IWizardOptions<T>);
 
-    public prompt(actionContext: IActionContext): Promise<void>;
-    public execute(actionContext: IActionContext): Promise<void>;
+    public prompt(): Promise<void>;
+    public execute(): Promise<void>;
 }
 
-export declare abstract class AzureWizardExecuteStep<T extends {}> {
+export declare abstract class AzureWizardExecuteStep<T extends IActionContext> {
```

Comment:
Why extend action context rather than having it as a member?

Reply:
It just felt simpler to settle on one context object as the base for all other context objects. Perhaps you made this comment because something like this wouldn't work?:
```typescript
const wizardContext = { ...actionContext, wizardProp = 'exampleProp' };

// This will work since `properties` is a reference to the `properties` object on `actionContext`
wizardContext.properties.exampleProp = 'exampleValue';

// this will be ignored since `suppressErrorDisplay` is not changed on the original `actionContext`
// If we had used `Object.assign` above, it would've worked. But I think people prefer spread operator over Object.assign
wizardContext.suppressErrorDisplay = true;
```

I just refactored `IActionContext` to hopefully alleviate that concern. This is what it'll look like now:
```typescript
const wizardContext = { ...actionContext };
wizardContext.telemetry.properties.exampleProp = 'exampleValue';
wizardContext.errorHandling.suppressDisplay = true;
```
(I also just like `.telemetry.properties` because it makes it more obvious what those properties are for)

Reply:
More wordy, but agreed better structured/behaved.